src/main/java/de/srendi/advancedperipherals/common/addons/refinedstorage/RSCraftJob.java

```diff
@@ -0,0 +1,250 @@
+        // It's cursed, but the only way currently
+        if (craftingTask != null) {
+            for (TaskStatus status : autocraftingComponent.getStatuses()) {
+                if (status.info().id() == craftingTask.info().id()) {
+                    this.craftingTask = status;
+                    break;
+                }
+            }
+        }
+        // The following is to get the TaskStatus from the issued task
+        if (futureTask == null || !futureTask.isDone() || craftingTask != null) {
+            return;
+        }
+        Optional<TaskId> optionalId;
+
+        try {
+            optionalId = futureTask.get();
+        } catch (InterruptedException | ExecutionException ex) {
+            AdvancedPeripherals.debug("Tried to get the task but the task was not done. Should be done", ex);
+            fireEvent(true, StatusConstants.UNKNOWN_ERROR);
+            return;
+        }
+
+        if (optionalId.isEmpty()) {
+            // This indicates that the second calculation was not successful. Well I guess. There is no java doc and I currently
+            // don't get an answer from the maintainer. So maybe we want to fire the crafting event
+            return;
+        }
+
+        TaskId id = optionalId.get();
+        for (TaskStatus status : autocraftingComponent.getStatuses()) {
+            if (status.info().id() == id) {
+                this.craftingTask = status;
+                // And only now we set that the crafting is started.
+                setStartedCrafting();
+                break;
+            }
+        }
+    }
+
+    @Override
+    protected void maybeCraft() {
+        if (startedCrafting || futureTask != null || futureCalculationResult == null || !futureCalculationResult.isDone()) {
+            return;
+        }
+
+        Optional<Preview> optionalPreview;
+
+        try {
+            optionalPreview = futureCalculationResult.get();
+        } catch (ExecutionException | InterruptedException ex) {
+            AdvancedPeripherals.debug("Tried to get preview, but preview calculation is not done. Should be done.", ex);
+            fireEvent(true, StatusConstants.UNKNOWN_ERROR);
+            return;
+        }
+
+        // TODO: I currently don't exactly know when the optional can be empty after the future is done. So I need to evaluate this.
+        if (optionalPreview.isEmpty()) {
+            AdvancedPeripherals.debug("preview optional is empty.", org.apache.logging.log4j.Level.ERROR);
+            fireEvent(true, StatusConstants.UNKNOWN_ERROR);
+            return;
+        }
+
+        Preview preview = optionalPreview.get();
+        this.preview = preview;
+
+        PreviewType previewType = preview.type();
+
+        if (previewType == PreviewType.MISSING_RESOURCES) {
+            fireEvent(true, StatusConstants.MISSING_ITEMS);
+            calculationNotSuccessful = true;
```

Comment:
```suggestion
            calculationNotSuccessful = true;
            fireEvent(true, StatusConstants.MISSING_ITEMS);
```

the order of `fireEvent` and `calculationNotSuccessful` is not same at here and below. Although they does not really matter I'd recommend make the order same to improve readability